crates/moonbuild/src/entry.rs

```diff
@@ -310,6 +312,75 @@ impl std::fmt::Display for TestResult {
     }
 }
 
+// (pkg's runnable artifacts, <mbt file name, array of (index, test_name)>) for blackbox whitebox internal test
+#[allow(clippy::type_complexity)]
+fn convert_moonc_test_info(
+    test_info_file: &Path,
+    pkg: &Package,
+    output_format: &str,
+    filter_file: Option<&String>,
+) -> anyhow::Result<
+    Vec<(
+        Option<PathBuf>,
+        IndexMap<String, IndexMap<u32, Option<String>>>,
+    )>,
+> {
+    let content = std::fs::read_to_string(test_info_file)
+        .context(format!("failed to read {}", test_info_file.display()))?;
+    let mut moonc_test_info = MooncGenTestInfo {
+        no_args_tests: IndexMap::new(),
+        with_args_tests: IndexMap::new(),
+    };
+    for line in content.split('\n') {
+        if let Ok(info) = serde_json_lenient::from_str::<MooncGenTestInfo>(line) {
+            moonc_test_info.no_args_tests.extend(info.no_args_tests);
```

Comment:
can we use `=` instead of `extend` here?

Reply:
there are three line of json(internal, blackbox, whitebox) in test_info.json, use `extend` to merge them into one `MooncGenTestInfo`